pkg/controller/directvolumemigration/rsync.go

```diff
@@ -792,6 +792,79 @@ func (t *Task) getRsyncOptions() []string {
 	return rsyncOpts
 }
 
+type PVCWithSecurityContext struct {
+	name               string
+	fsGroup            *int64
+	supplementalGroups []int64
+	seLinuxOptions     *corev1.SELinuxOptions
+
+	// TODO:
+	// add capabilities for dvm controller to handle case the source
+	// application pods is privileged with the following flags from
+	// PodSecurityContext and Containers' SecurityContext
+	// We need to
+	// 1. go through the pod's volume.
+	// 2. find the container where it is volume mounted.
+	//    i. if the container's fields are non-nil, use that
+	//    ii. if the container's fields are nil, use the pods fields.
+
+	// RunAsUser    *int64
+	// RunAsGroup   *int64
+	// RunAsNonRoot *bool
+}
+
+// Get fsGroup per PVC
+func (t *Task) getfsGroupMapForNamespace() (map[string][]PVCWithSecurityContext, error) {
+	pvcMap := t.getPVCNamespaceMap()
+	pvcSecurityContextMap := map[string][]PVCWithSecurityContext{}
+	for ns, _ := range pvcMap {
+		pvcSecurityContextMap[ns] = []PVCWithSecurityContext{}
+	}
+	for ns, pvcs := range pvcMap {
+		srcClient, err := t.getSourceClient()
+		if err != nil {
+			return nil, err
+		}
+		podList := &corev1.PodList{}
+		err = srcClient.List(context.TODO(), &k8sclient.ListOptions{Namespace: ns}, podList)
+		if err != nil {
+			return nil, err
+		}
+		for _, claimName := range pvcs {
```

Comment:
What if the claim is in [Lost](https://github.com/kubernetes/api/blob/cb95709d38de6602153fd4f5c05b036ba88256f5/core/v1/types.go#L592) status

Reply:
talked with shawn on slack we will take this issue up in 1.4.z